homeassistant/components/unifiprotect/camera.py

```diff
@@ -0,0 +1,184 @@
+"""Support for Ubiquiti's UniFi Protect NVR."""
+from __future__ import annotations
+
+from collections.abc import Callable, Generator, Sequence
+import logging
+from typing import Any
+
+from pyunifiprotect.api import ProtectApiClient
+from pyunifiprotect.data import Camera as UFPCamera
+from pyunifiprotect.data.devices import CameraChannel
+
+from homeassistant.components.camera import SUPPORT_STREAM, Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_BITRATE,
+    ATTR_CHANNEL_ID,
+    ATTR_FPS,
+    ATTR_HEIGHT,
+    ATTR_WIDTH,
+    DOMAIN,
+)
+from .data import ProtectData
+from .entity import ProtectDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_camera_channels(
+    protect: ProtectApiClient,
+) -> Generator[tuple[UFPCamera, CameraChannel, bool], None, None]:
+    """Get all the camera channels."""
+    for camera in protect.bootstrap.cameras.values():
+        if len(camera.channels) == 0:
+            _LOGGER.warning(
+                "Camera does not have any channels: %s (id: %s)", camera.name, camera.id
+            )
+            continue
+
+        is_default = True
+        for channel in camera.channels:
+            if channel.is_rtsp_enabled:
+                yield camera, channel, is_default
+                is_default = False
+
+        # no RTSP enabled use first channel with no stream
+        if is_default:
+            yield camera, camera.channels[0], True
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[Sequence[Entity]], None],
+) -> None:
+    """Discover cameras on a UniFi Protect NVR."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+    disable_stream = data.disable_stream
+
+    items = []
+    for camera, channel, is_default in get_camera_channels(data.api):
+        if channel.is_rtsp_enabled:
+            items.append(
```

Comment:
Looks like this and the else can be dropped since it's the same below

Reply:
I think I got what you mean, I cleaned it up so there are only two calls instead of 3 with an else.